marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -324,20 +353,34 @@ handleAction s (JavascriptAction action) = do
 handleAction settings (SimulationAction action) = do
   toSimulation (Simulation.handleAction settings action)
   case action of
-    ST.SetBlocklyCode -> do
-      mSource <- Simulation.editorGetValue
-      for_ mSource \source -> void $ query _blocklySlot unit (Blockly.SetCode source unit)
-      selectView BlocklyEditor
-    ST.EditHaskell -> selectView HaskellEditor
-    ST.EditJavascript -> selectView JSEditor
-    ST.EditActus -> selectView ActusBlocklyEditor
-    -- Replicate the state of unsavedChanges from the submodule/subcomponent into the MainFrame state
-    (ST.HandleEditorMessage (Monaco.TextChanged _)) -> do
-      hasUnsavedChanges <- queryCurrentEditorForUnsavedChanges
-      assign _hasUnsavedChanges hasUnsavedChanges
+    -- TODO: reimplement once we have the simulator in blockly and marlowe
+    ST.ViewAsBlockly -> pure unit
+    ST.EditSource -> do
+      flow <- use _workflow
+      case flow of
+        (Just lang) -> selectView $ selectLanguageView lang
+        _ -> pure unit
```

Comment:
I think it's more usual for us to use
```purescript
for_ flow \lang -> selectView $ selectLanguageView lang
```
or, depending on what you like,
```purescript
for_ flow (selectView <<< selectLanguageView)
```

Reply:
done